sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs

```diff
@@ -0,0 +1,114 @@
+        /// <param name="eventHubName">The name of the Event Hub, sometimes known as its path, that the sample should run against.</param>
+        /// <param name="tenantId">The Azure Active Directory tenant that holds the service principal.</param>
+        /// <param name="clientId">The Azure Active Directory client identifier of the service principal.</param>
+        /// <param name="secret">The Azure Active Directory secret of the service principal.</param>
+        ///
+        public async Task RunAsync(string fullyQualifiedNamespace,
+                                   string eventHubName,
+                                   string tenantId,
+                                   string clientId,
+                                   string secret)
+        {
+            int eventsPublished = 0;
+            int eventsRead = 0;
+
+            // Service principal authentication is a means for applications to authenticate against Azure Active
+            // Directory and consume Azure services. This is advantageous compared to using a connection string for
+            // authorization, as it offers a far more robust mechanism for transparently updating credentials in place,
+            // without an application being explicitly aware or involved.
+            //
+            // For this example, we'll take advantage of a service principal to publish and receive events.  To do so, we'll make
+            // use of the ClientSecretCredential from the Azure.Identity library to enable the Event Hubs clients to perform authorization
+            // using a service principal.
+
+            ClientSecretCredential credential = new ClientSecretCredential(tenantId, clientId, secret);
+
+            // To start, we'll publish a small number of events using a producer client.  To ensure that our client is appropriately closed, we'll
+            // take advantage of the asynchronous dispose when we are done or when an exception is encountered.
+
+            await using (var producerClient = new EventHubProducerClient(fullyQualifiedNamespace, eventHubName, credential))
+            {
+                using EventDataBatch eventBatch = await producerClient.CreateBatchAsync();
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Hello, Event Hubs!")));
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("The middle event is this one")));
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Goodbye, Event Hubs!")));
+
+                await producerClient.SendAsync(eventBatch);
+
+                eventsPublished = eventBatch.Count;
+                Console.WriteLine("The event batch has been published.");
+            }
+
+            // With our events published, we'll create a consumer client to read them.  We'll stop reading after we've received all events in the
+            // batch.
```

Comment:
We are actually reading the from the earliest position in the partition, correct? This is not the point of this sample anyway. Just thought someone might get confused.

Reply:
yeah, it's not ideal if someone is reusing a partition... I kind of shifted away from the "always do latest" to try and reduce complexity and show the intended majority case for `ReadEvents`, though you're right.   I don't have a good answer to this without adding complexity to the text.